### PR TITLE
samples: intel_s1000: update sample audio app for control from host

### PR DESCRIPTION
**Pull request by sathishkuttan**
This PR updates the 2-way audio sample application under `samples/boards/intel_s1000_crb/audio` to add a mechanism to control the target from a Linux host connected over USB.

A simple start/stop control is included in the sample for reference.

### CODE REVIEW

File: samples/boards/intel_s1000_crb/audio/audio_app.rst [hunk 1]
```diff
@@ -7,15 +7,17 @@ Overview
 ********
 
 This sample application demonstrates audio capture and playback on an
-Intel® S1000 CRB.
+Intel® Sue Creek S1000 CRB with control from a Linux host connected
```
Comment (dbkinder) [inline samples/boards/intel_s1000_crb/audio/audio_app.rst]: Don't use "Intel®" with a code name (e.g., Sue Creek).  The legal name is "Intel® Speech Enabling Developer Kit".  I'd recommend not mentioning Sue Creek.  For that matter S1000 isn't approved with an Intel® before it either.

I'd recommend just writing this name as ``Intel S1000 CRB``.

File: samples/boards/intel_s1000_crb/audio/audio_app.rst [hunk 2]
```diff
@@ -52,15 +60,32 @@ Pin 12 needs to be wired to Pin 29 on the 40-pin connector of the cable.
...
    :board:
    :goals: build
    :compact:
 
+Running
+*******
+
+Upon power up and downloading the application image (or booting the board
+with the image in flash), the application starts without user intervention
+and audio is transferred from the microphones to the connected connected
```
Comment (dbkinder) [inline samples/boards/intel_s1000_crb/audio/audio_app.rst]:
```suggestion
and audio is transferred from the microphones to the connected
```

File: samples/boards/intel_s1000_crb/audio/audio_app.rst [hunk 3]
```diff
@@ -52,15 +60,32 @@ Pin 12 needs to be wired to Pin 29 on the 40-pin connector of the cable.
...
+host and from the host to the audio codec output.
+
+If a Linux host is connected via USB to the Intel® Sue Creek S1000 board,
+the python script ``audio.py`` can be used to stop/[re]start audio transfer.
```
Comment (dbkinder) [inline samples/boards/intel_s1000_crb/audio/audio_app.rst]: Rewrite this as:
```
the python script ``audio.py`` can be used to start, stop, or restart audio transfer::

   $ cd $ZEPHYR_BASE/samples/boards/intel_s1000_crb/audio
   $ sudo -E python3 audio.py stop
   $ sudo -E python3 audio.py start

```
(Note the two colons ``::`` at the end of that line, and the indented command line examples)

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/boards/intel_s1000_crb/audio/audio_app.rst (hunk 1 above)
Comment (sathishkuttan) [reply samples/boards/intel_s1000_crb/audio/audio_app.rst]: Intel(R) Sue Creek S1000 is the now legal approved naming. Intel(R) S1000 is not correct, hence the change.
This is not the developer kit.

File: samples/boards/intel_s1000_crb/audio/audio_app.rst (hunk 2 above)
Comment (sathishkuttan) [reply samples/boards/intel_s1000_crb/audio/audio_app.rst]: corrected.

File: samples/boards/intel_s1000_crb/audio/audio_app.rst (hunk 3 above)
Comment (sathishkuttan) [reply samples/boards/intel_s1000_crb/audio/audio_app.rst]: used `code-block` instead. re-check.

File: samples/boards/intel_s1000_crb/audio/audio_app.rst [hunk 4]
```diff
@@ -7,15 +7,17 @@ Overview
...
 The application uses the following drivers
 
 - DMIC driver (For microphone audio capture)
 - I2S driver (For audio input from host,
   audio output to host and audio output to codec)
 - Codec driver
+- USB driver and USB HID stack
 
-Intel® S1000 is the master on the I2S interfaces.
+Intel® Sue Creek is the master on the I2S interfaces.
```
Comment (dbkinder) [inline samples/boards/intel_s1000_crb/audio/audio_app.rst]: Legal names can't be abbreviated, so this should be ``Intel® Sue Creek S1000``

File: samples/boards/intel_s1000_crb/audio/audio_app.rst [hunk 5]
```diff
@@ -25,22 +27,29 @@ The first 2 channels are buffered and forwarded to the host.
 Simultaneously, 2 channels of audio from the host are buffered and forwarded
 to the codec.
 This bidirectional forwarding of audio demonstrates a 2-way audio stream
-between S1000's microphones/audio line-out and the host.
+between the Sue Creek board's microphones/audio line-out and the host.
```
Comment (dbkinder) [inline samples/boards/intel_s1000_crb/audio/audio_app.rst]: Legal names can't be abbreviated, so this should be ``the Intel® Sue Creek S1000 board's``

File: samples/boards/intel_s1000_crb/audio/audio_app.rst (hunk 3 above)
Comment (dbkinder) [reply samples/boards/intel_s1000_crb/audio/audio_app.rst]: That works, thanks.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/boards/intel_s1000_crb/audio/audio_app.rst (hunk 1 above)
Comment (dbkinder) [reply samples/boards/intel_s1000_crb/audio/audio_app.rst]: Hmmm... OK.  That change isn't noted in the Intel trademark db (yet).

File: samples/boards/intel_s1000_crb/audio/audio_app.rst (hunk 4 above)
Comment (sathishkuttan) [reply samples/boards/intel_s1000_crb/audio/audio_app.rst]: sure. i don't mind using the looong name everywhere -- hard to read, but will make legal happy.

File: samples/boards/intel_s1000_crb/audio/audio_app.rst (hunk 5 above)
Comment (sathishkuttan) [reply samples/boards/intel_s1000_crb/audio/audio_app.rst]: recheck

**Review by dbkinder — APPROVED** [review]
doc change LGTM, thanks.

**Review by dbkinder — APPROVED** [review]
doc changes still LGTM, thanks.

**Review by dcpleung — APPROVED** [review]

**Pull request merged**